### ISSUE TITLE
Cancelling all jobs still leaves long running process

### ISSUE THREAD

**Issue by rijnhard** [node-schedule/node-schedule#548]
in the case of an application, shutdown node-schedule holds the application up because of long-running timers, which generally makes sense in cases except for an explicit shutdown.

fairly repeatable, especially in a mocha test
create a recurring schedule then

```javascript
import {
    scheduledJobs as scheduledCronJobs,
    cancelJob as cancelCronJob
} from 'node-schedule';
import wtf from 'wtfnode';
 
after(() => {
    wtf.dump();
    Object.keys(scheduledCronJobs).forEach(jobName => cancelCronJob(jobName));
    console.info(scheduledCronJobs); // prints empty
    wtf.dump(); // still has long running timer.
});
```

ideally, we should actually have a safe shutdown function for use in apps (not other libraries):
```javascript
function cancelAllJobs(triggered = false, cb = () => {}) {
    Object.keys(scheduledJobs).forEach(name => canceJob(name));

    if (triggered) {
        // cancel jobs that are still processing
    } else {
        // wait for triggered jobs to complete
    }
    
    cb();
}
```

**Comment by rijnhard** [comment]
If I have to speculate I think it has something to do with the calls to `prepareNextInvocation()` in `cancelInvocation()` which is called by `job.cancel()`;

**Comment by Caerbannog** [comment]
If I'm correct this was implemented in https://github.com/node-schedule/node-schedule/pull/583 and released in v2.1.0.

**Comment by rijnhard** [comment]
@Caerbannog yes it is. CLOSED
